### PR TITLE
Email 📬an application has been submitted to provider

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context
Currently, a support user must send the provider a notify email when an application has received two references and the 5 working day amendment period has expired. We want to send an automatic email to the provider notifying them when a new application has been submitted.

## Changes proposed in this pull request

This PR extends the ProviderMailer class and adds a new service `SendNewApplicationEmailToProvider` that is called when applications sent to a provider.

The service also audits these emails:
Screenshot 2020-01-30 at 11 49 03

## Guidance to review

There is an end-to-end test that covers the journey from when a candidate submits an application, the email sent to the provider. The reason I created a new service for the emails is that I believe we need to add other responsibilities such as auditing the emails. I̶'̶m̶ ̶p̶l̶a̶n̶n̶i̶n̶g̶ ̶t̶o̶ ̶a̶d̶d̶ ̶t̶h̶i̶s̶ ̶i̶n̶ ̶a̶ ̶s̶e̶p̶a̶r̶a̶t̶e̶ ̶P̶R̶.̶  I would like to receive some feedback on the `SendNewApplicationEmailToProvider` service name and the audit comment. Is it clear? Any suggestions for clearer name and message? 

## Link to Trello card

https://trello.com/c/09ms2J92/835-email-%F0%9F%93%AC-an-application-has-been-submitted-to-provider

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/provider_mailer/application_submitted.text.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Dear <%= @application.provider_user_name %>
+
+<%= @application.candidate_name %> submitted an application for <%= @application.course_name %> <%= @application.course_code %>. 
+
+Log in to Manage teacher training applications to respond:
+<%= provider_interface_application_choice_url(application_choice_id: @application.application_choice_id) %>
+
+We’ll reject the application on your behalf after 40 working days.   
+
+Contact us at becomingateacher@digital.education.gov.uk
```
Comment (chubberlisk) [inline app/views/provider_mailer/application_submitted.text.erb]:
```suggestion
Contact us at [becomingateacher@digital.education.gov.uk](mailto:becomingateacher@digital.education.gov.uk)
```

File: spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  include CourseOptionHelpers
+
+  scenario 'the provider receives email' do
+    given_i_am_a_provider_user_with_a_provider
+    and_a_candidate_submited_their_application
+
+    when_the_application_is_delivered_to_my_provider
+
+    then_i_should_receive_an_email_with_a_link_to_the_application
+    and_an_audit_comment_has_submitted
```
Comment (chubberlisk) [inline spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb]: :eyes: For the future PR?

Comment (C-gyorfi) [reply spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb]: Ouch, I'll remove this for now 😅

File: app/views/provider_mailer/application_submitted.text.erb (hunk 1 above)
Comment (C-gyorfi) [reply app/views/provider_mailer/application_submitted.text.erb]: Good catch!

File: spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      create(
+        :application_choice,
+        status: 'application_complete',
+        edit_by: Time.zone.today,
+        course_option: @course_option,
+        application_form:
+          create(
+            :completed_application_form,
+            submitted_at: Time.zone.today,
+),
```
Comment (tvararu) [inline spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb]:
```suggestion
          ),
```

There's another small misalignment later in this file but it doesn't bug me

File: spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb (hunk 2 above)
Comment (C-gyorfi) [reply spec/system/provider_interface/provider_receives_email_when_a_new_application_has_been_submitted_spec.rb]: In the end I did this in one PR

File: app/views/provider_mailer/application_submitted.text.erb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Dear <%= @application.provider_user_name %>
+
+<%= @application.candidate_name %> submitted an application for <%= @application.course_name %> <%= @application.course_code %>.
```
Comment (fofr) [inline app/views/provider_mailer/application_submitted.text.erb]: The course code must always be in brackets after the course name, eg:
`Mathematics (X100)` not `Mathematics X100`

File: app/services/send_applications_to_provider.rb [hunk 5]
```diff
@@ -4,6 +4,7 @@ def call
     GetApplicationFormsReadyToSendToProviders.call.each do |application_form|
       application_form.application_choices.each do |choice|
         SendApplicationToProvider.new(application_choice: choice).call
+        SendNewApplicationEmailToProvider.new(application_choice: choice).call
```
Comment (tijmenb) [inline app/services/send_applications_to_provider.rb]: Sorry to be late to the party here, but this probably belongs inside the `SendApplicationToProvider` class too!

**Pull request merged**